### PR TITLE
Wallet - bonding, new node stats component

### LINKED ISSUE #2321: Wallet - bonding, adding table with stats
image

We are adding table with node stats below main bonding table

These stats were alredy developed by us months ago(by Fouad) now we are just introducing this panel in different form

Panel is purely informative and you can't perform any action there

figma file: https://www.figma.com/file/KiYKZnWeefSALAe0j2dAPI/Nym-Wallet?node-id=123%3A3706

- [x] design
- [x] development

### PR DESCRIPTION
**Pull request by doums**
# Description

Closes: https://github.com/nymtech/nym/issues/2321

Notes:
this PR also
- disables the "bond more" button as this feature is not yet ready.
- fix some wrong displayed values on bonded node (displayed uNYM as NYM values)
- add a new tauri request to get the average uptime of a bonded node (displayed as "Routing score")

image

Blocked by #1528

### CODE REVIEW

File: nym-wallet/src/components/Bonding/NodeStats.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    }
+  };
+
+  const renderLegend = () => (
+    <Stack
+      alignItems="center"
+      maxWidth={200}
+      width={200}
+      sx={{
+        borderBottom: '1px solid #E8E9EB',
```
Comment (gala1234) [inline nym-wallet/src/components/Bonding/NodeStats.tsx]: We have this colour under the variable `theme.palette.nym.border.menu`

File: nym-wallet/src/components/Bonding/NodeStats.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    standbySetProbability,
+    operatorCost,
+    routingScore,
+  } = mixnode;
+  const theme = useTheme();
+  const data = [
+    { key: 'routingScore', value: routingScore },
+    { key: 'rest', value: 100 - routingScore },
+  ];
+  const colors = [theme.palette.success.main, '#E6E6E6'];
```
Comment (gala1234) [inline nym-wallet/src/components/Bonding/NodeStats.tsx]: Maybe, we could create a general variable for this colour in `nymPalette`, currently this colour is also used under the param `theme.palette.nym.nymWallet.background.greyStroke`, but only for light mode.

**Review by gala1234 — APPROVED** [review]

File: nym-wallet/src/components/Bonding/NodeStats.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    </Stack>
+    {typeof progressValue === 'number' ? (
+      <LinearProgressWithLabel value={progressValue} />
+    ) : (
+      <Typography>{value}</Typography>
+    )}
+  </Stack>
+);
+
+const StatDivider = () => <Box borderTop="1px solid" borderColor="rgba(141, 147, 153, 0.2)" my={1} />;
```
Comment (fmtabbara) [inline nym-wallet/src/components/Bonding/NodeStats.tsx]: MUI has a Divider component. You might be able to use that?

File: nym-wallet/src/components/Bonding/NodeStats.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+  } = mixnode;
+  const theme = useTheme();
+  const data = [
+    { key: 'routingScore', value: routingScore },
+    { key: 'rest', value: 100 - routingScore },
+  ];
+  const colors = [theme.palette.success.main, theme.palette.nym.nymWallet.chart.grey];
+
+  const getSetProbabilityLabel = (chance?: SelectionChance) => {
+    if (!chance) return 'Unknown';
```
Comment (fmtabbara) [inline nym-wallet/src/components/Bonding/NodeStats.tsx]: Won't the switch pick this up in the default case?

File: nym-wallet/src/components/Bonding/NodeStats.tsx [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+  const renderLegend = () => (
+    <Stack
+      alignItems="center"
+      maxWidth={200}
+      width={200}
+      sx={{
+        borderBottom: `1px solid ${theme.palette.nym.nymWallet.chart.grey}`,
+      }}
+    >
+      <Typography sx={{ color: (t) => t.palette.nym.text.muted }}>Routing score</Typography>
```
Comment (fmtabbara) [inline nym-wallet/src/components/Bonding/NodeStats.tsx]: When using sx prop in some cases (color, bgcolor, and some others) you don't need to access the theme this way. You can access the palette via a string `color: "nym.text.muted"`

File: nym-wallet/src/context/bonding.tsx [hunk 6]
```diff
@@ -165,6 +181,26 @@ export const BondingContextProvider = ({ children }: { children?: React.ReactNod
...
 
+  const getSetProbabilities = async (mixId: number) => {
+    let result;
+    try {
+      result = await getInclusionProbability(mixId);
+    } catch (e: any) {
+      Console.log(e);
+    }
+    return result;
+  };
```
Comment (fmtabbara) [inline nym-wallet/src/context/bonding.tsx]:
```
  const getSetProbabilities = async (mixId: number) => {
    try {
      const result = await getInclusionProbability(mixId);
      return result
    } catch (e: any) {
      Console.log(e);
    }
  };
```
Can do it this way?

File: nym-wallet/src/context/bonding.tsx [hunk 7]
```diff
@@ -165,6 +181,26 @@ export const BondingContextProvider = ({ children }: { children?: React.ReactNod
...
+  const getAvgUptime = async () => {
+    let result;
+    try {
+      result = await getMixnodeAvgUptime();
+    } catch (e: any) {
+      Console.log(e);
+    }
+    return result;
+  };
+
```
Comment (fmtabbara) [inline nym-wallet/src/context/bonding.tsx]: same as above?

**Review by fmtabbara — APPROVED** [review]

File: nym-wallet/src/context/bonding.tsx (hunk 6 above)
Comment (doums) [reply nym-wallet/src/context/bonding.tsx]: Yea, but doing this make ESLint not happy "Expected to return a value at the end of async arrow function. (eslint)"

**Pull request merged**